### PR TITLE
Do not write certs to disk

### PR DESCRIPTION
**Pull request by keyki**
Atm CBD still generates the keys, but it's not written to disk per stack anymore.

Periscope is not backward-compatible since earlier the keys were not encoded (on periscope side only, aka bug).

Next PR will be to get the 'cbd generated keys' from somewhere else other than the local disk.

### CODE REVIEW

**Comment by keyki** [comment]
Missing migration script for periscope

File: autoscale/src/main/java/com/sequenceiq/periscope/service/security/TlsSecurityService.java [hunk 1]
```diff
@@ -25,96 +19,30 @@
...
+        SecurityConfig securityConfig = cluster.getSecurityConfig();
+        if (securityConfig != null) {
+            return new TlsConfiguration(securityConfig.getClientKeyDecoded(), securityConfig.getClientCertDecoded(), securityConfig.getServerCertDecoded());
         }
-        return new TlsConfiguration(clientKeyPath.toString(), clientCertPath.toString(), serverCertPath.toString());
+        securityConfig = securityConfigRepository.findByClusterId(cluster.getId());
+        if (securityConfig == null) {
+            securityConfig = prepareSecurityConfig(cluster.getStackId());
+        }
+        return new TlsConfiguration(securityConfig.getClientKeyDecoded(), securityConfig.getClientCertDecoded(), securityConfig.getServerCertDecoded());
```
Comment (mhmxs) [inline autoscale/src/main/java/com/sequenceiq/periscope/service/security/TlsSecurityService.java]: "Return once" should be more readable

File: autoscale/src/main/java/com/sequenceiq/periscope/service/security/TlsSecurityService.java [hunk 2]
```diff
@@ -25,96 +19,30 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(TlsSecurityService.class);
```
Comment (mhmxs) [inline autoscale/src/main/java/com/sequenceiq/periscope/service/security/TlsSecurityService.java]: LOGGER not used any more.

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/client/KeyStoreUtil.java [hunk 3]
```diff
@@ -66,13 +64,12 @@ public static KeyStore createTrustStore(final String serverCertPath) throws Exce
         }
     }
 
-    private static Certificate loadCertificate(final String certPath) throws IOException, CertificateException {
-        File certificate = new File(certPath);
-        BufferedReader reader = null;
+    private static Certificate loadCertificate(final String cert) throws IOException, CertificateException {
```
Comment (mhmxs) [inline cloud-common/src/main/java/com/sequenceiq/cloudbreak/client/KeyStoreUtil.java]: Load in name doesn't represent the current behavior.

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/client/KeyStoreUtil.java [hunk 4]
```diff
@@ -88,13 +85,12 @@ private static Certificate loadCertificate(final String certPath) throws IOExcep
 
     }
 
-    private static KeyPair loadPrivateKey(final String clientKeyPath) throws IOException, InvalidKeySpecException, NoSuchAlgorithmException {
-        File privateKeyFile = new File(clientKeyPath);
-        BufferedReader reader = null;
+    public static KeyPair loadPrivateKey(final String clientKey) throws IOException, InvalidKeySpecException, NoSuchAlgorithmException {
```
Comment (mhmxs) [inline cloud-common/src/main/java/com/sequenceiq/cloudbreak/client/KeyStoreUtil.java]: Load in name doesn't represent the current behavior.

**Pull request merged**